scripts/test/Muon/duel_plot_maxent_pane/duel_plot_maxent_pane_model_test.py

```diff
@@ -0,0 +1,324 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2021 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+import unittest
+from unittest import mock
+from Muon.GUI.Common.ADSHandler.workspace_naming import (MAXENT_STR, get_maxent_workspace_name, get_raw_data_workspace_name,
+                                                         RECONSTRUCTED_SPECTRA)
+from Muon.GUI.Common.test_helpers.context_setup import setup_context
+from Muon.GUI.Common.utilities.algorithm_utils import create_empty_table
+from Muon.GUI.Common.muon_group import MuonGroup
+from Muon.GUI.Common.plot_widget.raw_pane.raw_pane_model import RawPaneModel
+from Muon.GUI.Common.plot_widget.data_pane.plot_data_pane_model import PlotDataPaneModel
+from Muon.GUI.FrequencyDomainAnalysis.plot_widget.duel_plot_maxent_pane.duel_plot_maxent_pane_model import DuelPlotMaxentPaneModel
+from mantidqt.utils.qt.testing import start_qapplication
+
+
+def gen_table():
+    tab = create_empty_table("table")
+    tab.addColumn('str', 'Group')
+    tab.addColumn('str', 'Detectors')
+    groups = ["fwd", "bwd"]
+    detectors_list = ["0,1,2,3,4", "7,8,9,10"]
+    for j, name in enumerate(groups):
+        detectors = detectors_list[j]
+        tab.addRow([name, detectors])
+    return tab
+
+
+@start_qapplication
+class DuelPlotMaxentPaneModelTest(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.context = setup_context()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.context.ads_observer = None
+
+    def setUp(self):
+        self.raw = mock.Mock(spec=RawPaneModel)
+        self.group = mock.Mock(spec=PlotDataPaneModel)
+
+        self.model = DuelPlotMaxentPaneModel(context=self.context, time_group_model=self.group, raw_model=self.raw)
+        test_group = MuonGroup(group_name="fwd", detector_ids=[1, 2, 3, 4, 5])
+        self.context.group_pair_context._groups = [test_group]
+        self.context.group_pair_context._pairs = []
+        self.context.group_pair_context._selected_groups = ["fwd"]
+        self.context.group_pair_context._selected_pairs = []
+        self.context.data_context.instrument = "MUSR"
```

Comment:
Looking through all the tests we only seem to only test for instrument MUSR. Do we need to test HIFI too?

Reply:
I just type that by default. The only reason for manual testing with HIFI is multiperiod data. However, in the tests we can make the MUSR data "multiperiod". So the instrument is irrelevant as we dont actually use it (beyond the name).